src/DeepCopy/DeepCopy.php

```diff
@@ -59,6 +61,7 @@ public function __construct($useCloneMethod = false)
     {
         $this->useCloneMethod = $useCloneMethod;
 
+        $this->addTypeFilter(new ArrayObjectFilter($this), new TypeMatcher(ArrayObject::class));
```

Comment:
Would it make sense to register it only on PHP 7.4+?

Reply:
I don't see any reason for not using it in PHP <7.4, do you?